src/v/cluster/topics_frontend.cc

```diff
@@ -1512,4 +1515,86 @@ topics_frontend::get_partition_state(model::ntp ntp) {
     co_return results;
 }
 
+std::vector<acl_resource>
+topics_frontend::do_describe_acls(const security::acl_binding_filter& filter) {
+    /*
+     * collapse common acls by pattern
+     */
+    absl::flat_hash_map<
+      security::resource_pattern,
+      std::vector<security::acl_entry>>
+      entries;
+
+    auto bindings = _authorizer.local().acls(filter);
+
+    for (const auto& binding : bindings) {
+        entries[binding.pattern()].push_back(binding.entry());
+    }
+
+    std::vector<acl_resource> resources;
+    for (const auto& entry : entries) {
+        acl_resource resource;
+
+        // resource pattern
+        resource.type = entry.first.resource();
+        resource.name = entry.first.name();
+        resource.pattern_type = entry.first.pattern();
+
+        // acl entries
+        for (auto& acl : entry.second) {
+            // ignore ephemeral_users
+            auto ephemeral_user = security::principal_type::ephemeral_user;
+            if (acl.principal().type() == ephemeral_user) {
+                continue;
+            }
+            acl_description desc{
+              .principal = acl.principal(),
+              .host = acl.host(),
+              .operation = acl.operation(),
+              .permission_type = acl.permission(),
+            };
+            resource.acls.push_back(std::move(desc));
+        }
```

Comment:
OUt of curiosity, do we skip ephemeral users because they cannot have ACLs or something?

Reply:
> OUt of curiosity, do we skip ephemeral users because they cannot have ACLs or something?

They can have ACLs, but they shouldn't be returned through the Kafka API.